pkg/api/norman/customization/kontainerdriver/actionhandler.go

```diff
@@ -154,30 +156,40 @@ func (lh ListHandler) LinkHandler(apiContext *types.APIContext, next types.Reque
 		rkeSysImages[k8sVersion] = *rkeSysImgCopy
 	}
 
-	// get system charts path
-	systemCatalog, err := lh.CatalogLister.Get("", utils.SystemLibraryName)
+	var catalogImageList *v1.ConfigMap
+	catalogImageList, err = lh.ConfigMapLister.Get(utils.GetCatalogImageCacheName(utils.SystemLibraryName), metav1.GetOptions{})
 	if err != nil {
-		return httperror.WrapAPIError(err, httperror.ServerError, "error getting system catalog")
+		return
 	}
-	systemCatalogHash := helmlib.CatalogSHA256Hash(systemCatalog)
-	systemCatalogChartPath := filepath.Join(helmlib.CatalogCache, systemCatalogHash)
 
-	var targetImages []string
+	var targetRkeSysImages []string
+	var targetSysCatalogImages []string
 	switch apiContext.ID {
 	case linuxImages:
-		targetImages, _, err = image.GetImages(systemCatalogChartPath, "", nil, []string{}, rkeSysImages, image.Linux)
+		targetRkeSysImages, _, err = image.GetImages("", "", nil, []string{}, rkeSysImages, image.Linux)
```

Comment:
Will targetRkeSysImages have anything since you were passing an empty path to `image.GetImages`? If it is empty I guess we don't have to deal with it.

Reply:
Yes it will have all the rkeSysImages and all of the mirrored rkeSysImages.  We only cache the chart images in the configmap not the system images.